scipy/stats/tests/test_continuous_basic.py

```diff
@@ -820,11 +820,22 @@ def test_broadcasting_in_moments_gh12192_regression():
     npt.assert_allclose(vals3, expected3, rtol=1e-8)
     assert vals3.shape == expected3.shape
 
+
 def test_kappa3_array_gh13582():
     # https://github.com/scipy/scipy/pull/15140#issuecomment-994958241
     shapes = [0.5, 1.5, 2.5, 3.5, 4.5]
     moments = 'mvsk'
-    res = np.array([[stats.kappa3.stats(shape, moments=moment) for shape in shapes]
-                    for moment in moments])
+    res = np.array([[stats.kappa3.stats(shape, moments=moment)
+                   for shape in shapes] for moment in moments])
     res2 = np.array(stats.kappa3.stats(shapes, moments=moments))
     npt.assert_allclose(res, res2)
+
+
+def test_kappa4_array_gh13582():
+    h = np.array([-1, -0.5, 0, 1, 2.5, 3.5, 4.5, -3])
+    k = np.array([-1, -0.5, 0, -2, 1, -1.5, 0, 3.5])
```

Comment:
Why this choice of parameters? It might be fine; I would just like to know the intent. 

If I were picking the numbers, I'd choose these to explore the different possible patterns of NaNs in `res` based on the logic. For instance, `np.isnan` of `res` when:
```python3
h = np.array([-1, -1/4, -1/4,    1, -1, 0])
k = np.array([ 1,    1,  1/2, -1/3, -1, 0])
```
is
```
array([[ True, False, False, False,  True, False],
       [ True, False,  True, False,  True, False],
       [ True, False,  True,  True,  True, False],
       [ True,  True,  True,  True,  True, False]])
```
The first three columns are showing the effect of h and k when `h < 0 and k >= 0`.
The next two columns are showing the effect of `k` when `k < 0`.
The last column shows what happens for the default `maxr`.

Reply:
> I would just like to know the intent.

I tried to get some different possible combinations based on the logic, `h < 0 and k >= 0`, `k <0 `, so that we can cover each case.

Reply:
Added a test.